libdevcrypto/Common.cpp

```diff
@@ -126,9 +117,40 @@ void dev::encryptECIES(Public const& _k, bytesConstRef _plain, bytes& o_cipher)
 
 void dev::encryptECIES(Public const& _k, bytesConstRef _sharedMacData, bytesConstRef _plain, bytes& o_cipher)
 {
-	bytes io = _plain.toBytes();
-	Secp256k1PP::get()->encryptECIES(_k, _sharedMacData, io);
-	o_cipher = std::move(io);
+	// interop w/go ecies implementation
+	auto r = KeyPair::create();
+	Secret z;
+	ecdh::agree(r.secret(), _k, z);
+	auto key = ecies::kdf(z, bytes(), 32);
+	bytesConstRef eKey = bytesConstRef(&key).cropped(0, 16);
+	bytesRef mKeyMaterial = bytesRef(&key).cropped(16, 16);
+	secp256k1_sha256_t ctx;
+	secp256k1_sha256_initialize(&ctx);
+	secp256k1_sha256_write(&ctx, mKeyMaterial.data(), mKeyMaterial.size());
+	bytes mKey(32);
+	secp256k1_sha256_finalize(&ctx, mKey.data());
+
+	auto iv = h128::random();
+	bytes cipherText = encryptSymNoAuth(SecureFixedHash<16>(eKey), iv, _plain);
+	if (cipherText.empty())
+		return;
+
+	bytes msg(1 + Public::size + h128::size + cipherText.size() + 32);
+	msg[0] = 0x04;
```

Comment:
Probably I would create a helper which creates a SEC.1 key.

SEC.1 (http://www.secg.org/sec1-v2.pdf) section 2.3.3 describes the rules.